tests/conftest.py

```diff
@@ -1,109 +1,27 @@
 HTTPS_SERVER_URL = "https://example.org"
 
 
-class RunNotify:
-    """A mitmproxy addon wrapping an event to notify us when the server is running."""
-
-    def __init__(self) -> None:
-        self.started = threading.Event()
-
-    def running(self) -> None:
-        self.started.set()
-
-
-class ProxyWrapper(threading.Thread):
-    """Runs an mitmproxy in a separate thread."""
-
-    def __init__(self, host: str, port: int, **kwargs: typing.Any) -> None:
-        self.host = host
-        self.port = port
-        self.options = kwargs
-        super().__init__()
-        self.notify = RunNotify()
-
-    def run(self) -> None:
-        # mitmproxy uses asyncio internally but the default loop policy
-        # will only create event loops for the main thread, create one
-        # as part of the thread startup
-        asyncio.set_event_loop(asyncio.new_event_loop())
-        opts = options.Options(
-            listen_host=self.host, listen_port=self.port, **self.options
-        )
-        pconf = proxy.config.ProxyConfig(opts)
-
-        self.master = DumpMaster(opts)
-        self.master.server = proxy.server.ProxyServer(pconf)
-        self.master.addons.add(self.notify)
-        self.master.run()
-
-    def join(self, timeout: float = None) -> None:
-        self.master.shutdown()
-        super().join()
-
-
-@pytest.fixture(scope="session")
-def cert_authority() -> trustme.CA:
-    return trustme.CA()
-
-
-@pytest.fixture()
-def ca_ssl_context(cert_authority: trustme.CA) -> ssl.SSLContext:
-    ctx = ssl.create_default_context()
-    cert_authority.configure_trust(ctx)
-    return ctx
-
-
-@pytest.fixture(scope="session")
-def example_org_cert(cert_authority: trustme.CA) -> trustme.LeafCert:
-    return cert_authority.issue_cert("example.org")
-
-
 @pytest.fixture(scope="session")
-def example_org_cert_path(example_org_cert: trustme.LeafCert) -> typing.Iterator[str]:
-    with example_org_cert.private_key_and_cert_chain_pem.tempfile() as tmp:
-        yield tmp
+def proxy_server() -> typing.Iterator[URL]:
+    proxy_host = "127.0.0.1"
+    proxy_port = 8080
 
-
-@pytest.fixture()
-def proxy_server(example_org_cert_path: str) -> typing.Iterator[URL]:
-    """Starts a proxy server on a different thread and yields its origin tuple.
-
-    The server is configured to use a trustme CA and key, this will allow our
-    test client to make HTTPS requests when using the ca_ssl_context fixture
-    above.
-
-    Note this is only required because mitmproxy's main purpose is to analyse
-    traffic. Other proxy servers do not need this but mitmproxy is easier to
-    integrate in our tests.
-    """
-    try:
-        thread = ProxyWrapper(PROXY_HOST, PROXY_PORT, certs=[example_org_cert_path])
-        thread.start()
-        thread.notify.started.wait()
-        yield (b"http", PROXY_HOST.encode(), PROXY_PORT, b"/")
-    finally:
-        thread.join()
+    with http_proxy_server(proxy_host, proxy_port) as proxy_server_fixture:
```

Comment:
(Nit) How about `as proxy_url`, since it seems the context manager returns an URL tuple?

Reply:
Renamed